### PR TITLE
CI, STYLE sync flake8 versions

### PR DESCRIPTION
**Pull request by MarcoGorelli**
There's been a couple of recent CI failures due to differences in flake8 versions: #40481, #40457

CI failures due to style checks are really annoying, so I think it's worth adding a check to ensure they're in-sync. There's already a hook with `pyyaml` as additional dependency, so that environment will be reused for this check.

### CODE REVIEW

File: environment.yml [hunk 1]
```diff
@@ -20,7 +20,7 @@ dependencies:
   # code checks
   - black=20.8b1
   - cpplint
-  - flake8
+  - flake8=3.9.0
```
Comment (jreback) [inline environment.yml:23]: can we have 1 build where this is unpinned?

Comment (MarcoGorelli) [reply environment.yml:23]: Sure, but there's the risk that there's be another annoying CI failure like #40457 

If there's a new flake8 release, this'll get bumped up anyway by the weekly autoupdate job

Comment (jreback) [reply environment.yml:23]: yeah this is ok then.

**Comment by jreback** [comment]
can you merge master and merge on green.

**Comment by jreback** [comment]
thanks @MarcoGorelli

**Pull request merged**